src/helper.jl

```diff
@@ -165,3 +165,181 @@ function writetable(filename::AbstractString; rewrite::Bool=false, kw...)
     writexlsx(filename, xf, rewrite=rewrite)
     nothing
 end
+
+"""
+    emptyfile(sheetname::AbstractString="")
+
+Returns an empty, writable `XLSXFile` with 1 worksheet.
+
+`sheetname` is the name of the worksheet, defaults to `Sheet1`.
+"""
+function emptyfile(sheetname::AbstractString="")
+    xf = open_default_template()
+
+    if sheetname != ""
+        rename!(xf[1], sheetname)
+    end
+
+    return xf
+end
+
+function Base.open(f::Function, filename::AbstractString, sheetname::AbstractString="")
+    xf = open_default_template()
+
+    if sheetname != ""
+        rename!(xf[1], sheetname)
+    end
```

Comment:
So `open` has two main purposes: reading/writing. It looks like this code only handles creating of a new file. To handle both I think you could do:
```julia
if isfile(filename)
    xf = openxlsxtemplate(filename)
    sheet = if !isempty(sheetname)
        if hassheet(xf, sheetname)
            getsheet(xf, sheetname)
        else
            # create a sheet?
        end
    else
         xf[1]
    end
else
    xf = open_default_template()
    sheetname != "" && rename!(xf[1], sheetname)
end
```
I'm not sure if it makes sense to pass in the `Workbook` or `Worksheet` to `f`.

Reply:
Hmm, I'm starting to think this shouldn't interact with sheets at all